geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -89,8 +92,31 @@ object AccumuloDataStoreFactory {
     val instance = instanceIdParam.lookup[String](params)
     val user = userParam.lookup[String](params)
     val password = passwordParam.lookup[String](params)
+    val keytabPath = keytabPathParam.lookup[String](params)
+
+    var usingKerberos = false
+
+    // Build authenticaton token according to how we are authenticating
+    val authToken : AuthenticationToken = if (password != null ) {
+      new PasswordToken(password.getBytes("UTF-8"))
+    } else if (keytabPath != null && !useMock) {
+      usingKerberos = true
+      // Get ctor for principal, keytab path and replace current user
+      val ctor = Class.forName("org.apache.accumulo.core.client.security.tokens.KerberosToken")
+        .getConstructor(classOf[String], classOf[java.io.File], classOf[Boolean])
+      try {
+        (ctor.newInstance(user, new java.io.File(keytabPath), java.lang.Boolean.TRUE)).asInstanceOf[AuthenticationToken]
+      } catch {
+        // Remove reflection exception wrapper
+        case ite: InvocationTargetException => throw ite.getCause
+      }
+    } else {
+      // Will get here only if using mock kerberos thanks to parameter validation.
+      // MockInstance doesn't seem to support Kerberos, so make up a password
+      // Blank password is for the root user
+      new PasswordToken("".getBytes("UTF-8"))
```

Comment:
could we switch the logic here slightly, and move the `!useMock` inside the `else if (keytabPath != null)`, then make the final else throw an exception of some kind since no authentication was provided (previously it would throw an NPE on line 93, which wasn't ideal, but this change would mask that)
edit: I realize that it shouldn't be in that state due to canProcess, but I think it would still be better to be defensive here